service/lib/agama/dbus/storage/proposal_settings_conversion/from_dbus.rb

```diff
@@ -0,0 +1,165 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2023] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage/encryption_method"
+require "y2storage/pbkd_function"
+require "agama/storage/proposal_settings"
+require "agama/storage/proposal_settings_reader"
+require "agama/storage/space_settings"
+require "agama/dbus/storage/volume_conversion"
+
+module Agama
+  module DBus
+    module Storage
+      module ProposalSettingsConversion
+        # Proposal settings conversion from D-Bus format.
+        class FromDBus
+          # @param dbus_settings [Hash]
+          # @param config [Agama::Config]
+          def initialize(dbus_settings, config:)
+            @dbus_settings = dbus_settings
+            @config = config
+          end
+
+          # Performs the conversion from D-Bus format.
+          #
+          # @return [Agama::Storage::ProposalSettings]
+          def convert
+            settings = Agama::Storage::ProposalSettingsReader.new(config).read
+
+            settings.tap do |target|
+              dbus_settings.each do |dbus_property, dbus_value|
+                converter = CONVERSIONS[dbus_property]
+                # FIXME: likely ignoring the wrong attribute is not the best
```

Comment:
at least log it, so we can catch typos in logs.

Reply:
We are going to create a follow-up issue to validate data from both D-Bus and config file.